src/shared/Core/ApplicationBase.cs

```diff
@@ -91,7 +91,7 @@ public static string GetEntryApplicationPath()
 
         public static string GetInstallationDirectory()
         {
-            return Path.GetDirectoryName(Environment.GetCommandLineArgs()[0]);
+            return AppContext.BaseDirectory;
```

Comment:
It is fantastic that you tested all those scenarios _and wrote about it_ in the commit message. Otherwise I would have had concerns that this break things left and right. But that testing builds confidence in the correctness of this change!